ai-player.js

```diff
@@ -235,10 +235,46 @@ function createAiPlayer(game, options) {
         if (state.state.name == stateNames.FINAL_ACTION_RESPONSE) {
             return false;
         }
+
+        // Challenge if somebody claims to have role that was revealed 3 times or we have the rest of them
+        var usedRoles = countRevealedRoles(state.state.action);
+        for (var i = 0; i < aiPlayer.influence.length; i++) {
+            if (!aiPlayer.influence[i].revealed && aiPlayer.influence[i].role === state.state.action) {
+                usedRoles++;
+            }
+        }
+        if (usedRoles === 3) {
+            return true;
+        }
+
+        // Challenge if you're being assassinated, it's your last influence and all contessas have been used
+        if (state.state.action === 'assassinate' && state.players[state.playerIdx].influenceCount === 1) {
+            var contessas = countRevealedRoles('contessa');
+            if (contessas === 3) {
+                return true;
+            }
+            // Challenge when there are no contessas that were not claimed
```

Comment:
This line took me a while to understand. Could you update the comment to explain that you are counting all contessas that have been revealed or claimed?